### DISCUSSION TITLE
8BitDo controller on Raspberry Pi

### DISCUSSION THREAD

**Discussion by stephenpapierski** [winder/Universal-G-Code-Sender#2137] [Help]
I have UGS Platform running on a Raspberry Pi 4 loaded the the Raspberry Pi OS. I purchased an 8BitDo Lite 2 to enhance my control. It claims to be Raspberry Pi compatible, but everything I can find on setting it up points to RetroPi instead of Raspberry Pi OS. I'm able to successfully pair the Bluetooth controller, but when I select "Enable Joystick", nothing seems to happen. From watching a video, I understand that the key presses should light up the corresponding key in that configuration menu. I seem to get nothing. Any guidance? Do I need to install some drivers or should I be looking at a different type of controller?

**Comment by breiler** [comment]
The controller should be compatible with the gamepad/joystick library we are using according to the `gamecontrollerdb.txt`.
https://github.com/winder/Universal-G-Code-Sender/blob/759b21f85495a11de32df69d0217257235d64301/ugs-platform/ugs-platform-plugin-joystick/src/main/resources/com/willwinder/ugs/nbp/joystick/gamecontrollerdb.txt#L1014

There might be different versions of the controller that causes it to not be recognized by UGS. You can verify this by using one of these tools: https://github.com/gabomdq/SDL_GameControllerDB#mapping-tools
The ID of your controller should map one of the ones in the `gamecontrollerdb.txt`.

If it is recognized in any of the mapping tools it might be a problem with UGS. In that case you can post your log file: `~/.ugsplatform/var/log/messages.log`

**Reply by stephenpapierski** [reply]
I finally got back to trying to get the controller to work. Thanks for the lead on the log location, that ended up helping out. It seems my problem was pairing the controller after starting UGS. In that case, it never picks up on it. If the controller is paired before starting UGS, it grabs it.

For my use case, I auto start UGS (small computer running a small touch screen), then I would turn on the game controller later. Is there a way to get UGS to scan for controllers after startup? Leaving it plugged into USB seems like the solution I may go with.
